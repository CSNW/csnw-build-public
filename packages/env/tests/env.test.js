const { join } = require('path');
const { Files } = require('@csnw-build/pipeline');

process.env.CSNW_BUILD_TEST_A = 'aaa';
process.env.CSNW_BUILD_TEST_B = 'bb';
process.env.CSNW_BUILD_TEST_C = 'c';

const env = require('../')();
const cwd = join(__dirname, 'fixtures');

test('pipeline', async () => {
  const process = await env.pipeline({
    cwd,
    sourcemaps: true,
    cache: false
  });
  const files = await Files.from('*.{js,html}', { cwd });

  const result = await process(files);
  expect(result).toMatchSnapshot();
});
