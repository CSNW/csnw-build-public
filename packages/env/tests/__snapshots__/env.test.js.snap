// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pipeline 1`] = `
Object {
  "a.js": Object {
    "baseDir": "C:/dev/csnw-build/packages/env/tests/fixtures",
    "contents": "var a = \\"aaa\\";
var b = \\"bb\\";
var unknown = \\"\\";
var c = \\"c\\";
",
    "cwd": "C:/dev/csnw-build/packages/env/tests/fixtures",
    "path": "C:/dev/csnw-build/packages/env/tests/fixtures/a.js",
    "sourceMap": Object {
      "file": "a.js",
      "mappings": "AAAA,QAAQ,KAA6B;QAC7B,IAA6B;cACvB,EAAmC;QACzC,GAA6B;",
      "names": Array [],
      "sources": Array [
        "a.js",
      ],
      "sourcesContent": Array [
        "var a = process.env.CSNW_BUILD_TEST_A;
var b = process.env.CSNW_BUILD_TEST_B;
var unknown = process.env.CSNW_BUILD_TEST_UNKNOWN;
var c = process.env.CSNW_BUILD_TEST_C;
",
      ],
      "version": 3,
    },
  },
  "b.html": Object {
    "baseDir": "C:/dev/csnw-build/packages/env/tests/fixtures",
    "contents": "<h1>aaa</h1>
<p>
  bb
  <span></span>
  c
</p>",
    "cwd": "C:/dev/csnw-build/packages/env/tests/fixtures",
    "path": "C:/dev/csnw-build/packages/env/tests/fixtures/b.html",
    "sourceMap": Object {
      "file": "b.html",
      "mappings": "AAAA,IAAI,GAAmB;;EAErB,EAAmB;QACb,AAAyB;EAC/B,CAAmB;",
      "names": Array [],
      "sources": Array [
        "b.html",
      ],
      "sourcesContent": Array [
        "<h1>%CSNW_BUILD_TEST_A%</h1>
<p>
  %CSNW_BUILD_TEST_B%
  <span>%CSNW_BUILD_TEST_UNKNOWN%</span>
  %CSNW_BUILD_TEST_C%
</p>",
      ],
      "version": 3,
    },
  },
}
`;
